atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/cleaner/ImmutableTimestampSupplierProgressMonitor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cleaner;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Supplier;
+import com.google.common.collect.Lists;
+import com.palantir.lock.LockClient;
+
+final class ImmutableTimestampSupplierProgressMonitor implements Supplier<Long> {
+    private static final Logger log = LoggerFactory.getLogger(ImmutableTimestampSupplierProgressMonitor.class);
+
+    private static final long TIME_TO_WARN_HR = 1L;
+    private static final long TIME_TO_ERROR_HR = 24L;
+    private static final String LONG_RUNNING_TRANSACTION_ERROR_MESSAGE = "Immutable timestamp has not been updated for"
+            + " [{}] hour(s) for LockClient [{}]. This indicates to a very long running transaction.";
+
+    private final Supplier<Long> supplier;
+
+    private final List<ImmutableTimestampMonitor> monitors;
+
+    private ImmutableTimestampSupplierProgressMonitor(Supplier<Long> supplier,
+                                                      List<ImmutableTimestampMonitor> monitors) {
+        this.supplier = supplier;
+        this.monitors = monitors;
+    }
+
+    static Supplier<Long> createWithDefaultMonitoring(Supplier<Long> supplier, LockClient lockClient) {
+        List<ImmutableTimestampMonitor> defaultMonitors = Lists.newArrayList();
+        defaultMonitors.add(new ImmutableTimestampMonitor(TimeUnit.HOURS.toMillis(TIME_TO_WARN_HR),
+                () -> log.warn(LONG_RUNNING_TRANSACTION_ERROR_MESSAGE, TIME_TO_WARN_HR, lockClient.getClientId())));
+
+        defaultMonitors.add(new ImmutableTimestampMonitor(TimeUnit.HOURS.toMillis(TIME_TO_ERROR_HR),
```

Comment:
I think that would move duplication inside `ImmutableTimestampMonitor` as SLF4J doesn't support dynamic logging level (I could pass lambda with params, but that would make it even worse I think). Also this way it is easy to test.